### PR TITLE
add OnSwipe class and the ability to set and onSwipe and onClick

### PR DESCRIPTION
**Pull request by jafu888**

### CODE REVIEW

**Comment by rodrigomartindMeli** [comment]
I like to see this PR, it is something that will be very useful for me.

**Review by rodrigomartindMeli — APPROVED** [review]
Reward for you 🍎🍎🍎🍎🍎

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionScene.java [hunk 1]
```diff
@@ -490,12 +490,41 @@ public boolean applyViewTransition(int viewTransitionId, MotionController motion
...
+            TransitionOnClick click = new TransitionOnClick(this, id, action);
+            mOnClicks.add(click);
+        }
+
+        public void removeOnClick(int id) {
+            TransitionOnClick toRemove = null;
+            for (TransitionOnClick onClick : mOnClicks) {
+                if (onClick.mTargetId == id) {
+                    toRemove = onClick;
+                    return;
```
Comment (MGaetan89) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionScene.java:514]: Did you mean to use `break` here?

Comment (jafu888) [reply constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionScene.java]: Yes Thanks for checking!

**Review by camaelon — APPROVED** [review]

**Pull request merged**